src/binkmovie/bink_patches.c

```diff
@@ -0,0 +1,225 @@
+
+    WWMouseClass__Show_Mouse(WWMouse);
+    GScreenClass__Flag_To_Redraw(&MouseClass_Map, 2);
+    WWKeyboardClass__Clear(WWKeyboard);
+	
+	return TRUE;
+}
+
+//
+// Intercept to the games Play_Movie which checks if the Bink video file is 
+// available, falling back to VQA if not.
+//
+void __fastcall Play_Movie_Intercept(char *filename, int theme, bool a3, bool stretch, bool hidden)
+{
+	static char filename_buffer[32];
+	strncpy(filename_buffer, filename, 32);
+	
+	// Find the location of the file extension indentifier.
+    char *movie_name = strchr((char *)filename_buffer, '.');
+	
+	// Unexpected filename format passed in.
+    if (!movie_name) {
+        WWDebug_Printf("Invalid movie filename \"%s\"!\n", filename_buffer);
+        return;
+    }
+
+	// Insert a null-char where the "." was. This will give us the actual
+	// movie name without the extension, allowing us to rebuild them.
+    *movie_name = '\0';
+
+    char *upper_filename = strupr((char *)filename_buffer);
+
+    char bink_buffer[32-4];
+    sprintf(bink_buffer, "%s.BIK", upper_filename);
+	
+    char vqa_buffer[32-4];
+    sprintf(vqa_buffer, "%s.VQA", upper_filename);
+	
+    CCFileClass binkfile;
+    CCFileClass vqafile;
+	
+	CCFileClass__CCFileClass(&binkfile, bink_buffer);
+	CCFileClass__CCFileClass(&vqafile, vqa_buffer);
+	
+	BOOL bink_available = CCFileClass__Is_Available(&binkfile, FALSE);
+	BOOL vqa_available = CCFileClass__Is_Available(&vqafile, FALSE);
+	
+	// If the movie exists as a .BIK, if so, play as Bink!
+	if (bink_available) {
+		WWDebug_Printf("Play_Movie \"%s\" as Bink!\n", upper_filename);
+		if (Play_Movie_As_Bink(bink_buffer, theme, a3, stretch, hidden)) {
+			CCFileClass__Destroy(&binkfile);
+			CCFileClass__Destroy(&vqafile);
+			return;
+		}
+	}
+		
+	// The movie did not exist as a .BIK or failed to play, attempt to play the .VQA.
+    if (vqa_available) {
+        WWDebug_Printf("Play_Movie \"%s\" as VQA!\n", upper_filename);
+        Play_Movie(vqa_buffer, theme, a3, stretch, hidden);	// Call the game Play_Movie.
+		
+    } else {
+		WWDebug_Printf("Failed to play movie \"%s\"!\n", upper_filename);
+	}
+		
+	CCFileClass__Destroy(&binkfile);
+	CCFileClass__Destroy(&vqafile);
+}
+
+
+BOOL BinkMovieAdded = FALSE;
+
+/*
+static bool Bink_Movie_Available(char *movie_name)
+{
+    char buffer[32-4];
+    sprintf(buffer, "%s.BIK", movie_name);
+	
+    CCFileClass file;
+	CCFileClass__CCFileClass(&file, buffer);
+	
+	BOOL available = CCFileClass__Is_Available(&file, FALSE);
+	
+	CCFileClass__Destroy(&file);
+	
+	return available;
+}
+
+
+static bool VQA_Movie_Available(char *filename)
+{
+    char buffer[32-4];
+    sprintf(buffer, "%s.VQA", movie_name);
+	
+    CCFileClass file;
+	CCFileClass__CCFileClass(&file, buffer);
+	
+	BOOL available = CCFileClass__Is_Available(&file, FALSE);
+	
+	CCFileClass__Destroy(&file);
+	
+	return available;
+}
+
+
+from: 00563B7E
+to: 00563BB6
+
+if (Bink_Movie_Available()) {
+	BinkMovieAdded = TRUE;
+}
+*/
```

Comment:
Are these notes / comments relevant anymore? As far as I can tell, `Play_Movie_Intercept` contains the code for these availability checker functions so it's not needed to leave them in comments.